src/full_client.toit

```diff
@@ -668,6 +668,9 @@ class Session_:
     while true:
       candidate := next-packet-id_
       next-packet-id_ = (candidate + 1) & ((1 << PublishPacket.ID-BIT-SIZE) - 1)
+
+      // Message ID 0 is reserved as invalid.
+      if next-packet-id_ == 0: next-packet-id_ = 1
```

Comment:
I think you'll need a comment at the definition of next-packet-id_ too. If this starts at zero, you'll be out of luck.

Reply:
Changed it to skip candidates that are equal to 0.